docs/src/guide/queries_iteration.md

```diff
@@ -0,0 +1,292 @@
+> Therefore, it is advisable to avoid applying the `len` function
+> on queries in "hot" code.
+
+## Iterating over queries
+
+As we have seen, we can iterate over queries using a for loop.
+Here, the control variable ("entity") is an {{< api EntityAccessor >}} 
+object, i.e., not technically an {{< api Entity >}}, which is
+merely an identifier of an entity. Instead, the `EntityAccessor`
+directly provides methods to get, set, and check the existence
+of components, so that we do not need to call the world's 
+methods for this.
+
+```mojo {doctest="guide_queries_iteration"}
+for entity in world.query[Position]():
+    pos = entity.get_ptr[Position]()
+    print(
+        "Entity at position: (" 
+        + String(pos[].x) + ", " + String(pos[].y) + ")"
+    )
+    if entity.has[Velocity]():
+        # Reset position and velocity
+        entity.set(Position(0, 0), Velocity(0, 0))
+```
+
+> [!Note] 
+> The `EntityAccessor` is a temporary object that is
+> created for each iteration. Therefore, should not be
+> stored in a container.
+
+> [!Note] 
+> Adding or removing entities or components to/from entities
+> would invalidate the iterator. Therefore, the world is 
+> locked during the iteration. This means that the forbidden
+> operations will raise an exception during iterations.
+
+## Applying functions to entities in queries
+
+We may want to apply a certain operation to all entities
+that have certain components. This can be achieved with 
+the {{< api World.apply apply >}} method. This method
+iterates over all entities conforming to a query and
+calls the provided function with the entities as arguments.
+The function must take a {{< api MutableEntityAccessor >}} 
+as its only argument.
+
+For example, if we want to apply a function that moves all entities
+with a `Position` and a `Velocity` component, we can do this as follows:
+
+```mojo {doctest="guide_queries_iteration"}
+# Define the move function
+fn move(entity: MutableEntityAccessor) capturing:
+    try:
+        move_pos = entity.get_ptr[Position]()
+        move_vel = entity.get_ptr[Velocity]()
+        move_pos[].x += move_vel[].dx
+        move_pos[].y += move_vel[].dy
+    except:
+        # We could do proper error handling here
+        # but for now, we just ignore the error
+        pass
+
+# Apply the move function to all entities with a position and a velocity
+world.apply[move](world.query[Position, Velocity]())
+```
+
+> [!Note] 
+> Currently, the applied operation can not raise exceptions.
+> Therefore, we need to catch exceptions in the function
+> itself. This is due to current limitations of Mojo and
+> will be changed as soon as possible.
+
+> [!Caution]
+> The world is locked during the iteration, and 
+> accessing variables outside a locally defined
+> function is an immature feature in Mojo. Do not
+> attempt to access the `world` from inside the operation.
+
+## Application of vectorized functions
```

Comment:
As this is quite an advanced feature, I would (slightly) tend to move it to a separate chapter.